openehr-rm/src/main/java/com/nedap/archie/rminfo/UpdatedValueHandler.java

```diff
@@ -35,6 +40,44 @@
         if(parent instanceof CodePhrase) {
             return fixCodePhrase(rmObject, archetype, pathOfParent);
         }
+        if (parent instanceof DvQuantity) {
+            return fixDvQuantity(rmObject, archetype, pathOfParent);
+        }
+
+        return new HashMap<>();
+    }
+
+    private static Map<String, Object> fixDvQuantity(Object rmObject, Archetype archetype, String pathOfParent) {
+        try {
+            Map<String, Object> result = new HashMap<>();
+
+            RMPathQuery rmPathQuery = new RMPathQuery(pathOfParent);
+            DvQuantity quantity = rmPathQuery.find(ArchieRMInfoLookup.getInstance(), rmObject);
+
+            OperationalTemplate template = (OperationalTemplate) archetype;
+
+            CAttribute units = template.getDefinition().itemAtPath(pathOfParent + "/units");
+            CAttribute precision = template.getDefinition().itemAtPath(pathOfParent + "/precision");
+
+            if (units.getChildren().size() != 1 || precision.getChildren().size() != 1)
+                return result; // Only fix if there is 1 unit and 1 precision, don't assume anything otherwise
+
+            // Fix units
+            CString cString = (CString) units.getChildren().get(0);
+            String assumedUnitValue = cString.getAssumedValue();
```

Comment:
assumed value won't work, that's an explicitly set thing in ADL. You need the actual first+only constraint here, need to check if it's not a regular expression (starts+ends with / or ^), and set it.